src/multivariate/mvnormalcanon.jl

```diff
@@ -112,10 +113,14 @@ function MvNormalCanon(h::AbstractVector{T}, J::P) where {T<:Real, P<:AbstractPD
     MvNormalCanon{eltype(hh),typeof(JJ),typeof(hh)}(JJ \ hh, hh, JJ)
 end
 
+MvNormalCanon(μ::AbstractVector{<:Real}, h::AbstractVector{<:Real}, J::AbstractSparseMatrix{<:Real}) = MvNormalCanon(μ, PDSparseMat(J))
```

Comment:
These definitions would have to go behind a `isdefined` flag to avoid breaking builds without the sparse libraries.